IBAnimatable/Navigator.swift

```diff
@@ -74,3 +75,43 @@ extension Navigator: UINavigationControllerDelegate {
     }
   }
 }
+
+// MARK: - tabbar controller delegate
+extension Navigator: UITabBarControllerDelegate {
+
+  // MARK: - animation controller
+  public func tabBarController(_ tabBarController: UITabBarController,
+                               animationControllerForTransitionFrom fromVC: UIViewController,
+                               to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+
+    interactiveAnimator?.connectGestureRecognizer(to: toVC)
+
+    guard let viewControllers = tabBarController.viewControllers,
+      let fromVCIndex = viewControllers.index(of: fromVC),
+      let toVCIndex = viewControllers.index(of: toVC) else {
+        return nil
+    }
+    if toVCIndex < fromVCIndex {
+      animator?.transitionDuration = transitionDuration
+      return animator
+    } else {
+      // Use the reverse animation
+      if let reverseTransitionAnimationType = animator?.reverseAnimationType {
+        return AnimatorFactory.makeAnimator(transitionAnimationType: reverseTransitionAnimationType, transitionDuration: transitionDuration)
+      }
+      animator?.transitionDuration = transitionDuration
+      return animator
+    }
+  }
+
+  // MARK: - interaction controller
+  public func tabBarController(_ tabBarController: UITabBarController,
+                               interactionControllerFor animationController: UIViewControllerAnimatedTransitioning)
+    -> UIViewControllerInteractiveTransitioning? {
+      if let interactiveAnimator = interactiveAnimator, interactiveAnimator.interacting {
+        return interactiveAnimator
+      } else {
```

Comment:
Yes it's a copy paste like always, I am a copy past coder 😉 

here I think I could do
```swift
return interactiveAnimator?.interacting
```
no if else